app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala

```diff
@@ -316,32 +328,13 @@ class SectionRenderingService(
     val addAnotherQuestion: Html =
       new components.GovukRadios(govukErrorMessage, govukFieldset, govukHint, govukLabel)(radios)
 
-    val limitReached: AddToListLimitReached = bracket.source.limit
-      .flatMap { limit =>
-        val maybeMax: Option[BigDecimal] =
-          formModelOptics.formModelVisibilityOptics.evalAndApplyTypeInfoFirst(limit.repeatsMax).numberRepresentation
-
-        limit.field match {
-          case info @ IsInformationMessage(InformationMessage(infoType, infoText)) =>
-            maybeMax.map { maxRepeatsBigDecimal =>
-              if (maxRepeatsBigDecimal.toInt <= bracket.iterations.size) {
-                AddToListLimitReached.Yes(
-                  HtmlFormat.fill(
-                    List(
-                      htmlForInformationMessage(info, infoType, infoText),
-                      html.form.snippets.hidden(formComponent.id.value, "1")
-                    )
-                  )
-                )
-              } else {
-                AddToListLimitReached.No
-              }
-            }
+    val evalRepeatsUntil = bracket.source.repeatsUntil
```

Comment:
Any usage of `bracket.source` like `bracket.source.repeatsUntil` seems like a bug. `repeatsUntil` includeIf should be able to refer to fields in current iteration etc.